trusted-types/eval-no-csp-no-tt.html

```diff
@@ -37,8 +37,20 @@
     assert_equals(new Function(p.createScript('return 1+1'))(), 2);
   }, "Function constructor of TrustedScript works.");
 
+  test(t => {
+    assert_equals(new Function(p.createScript('val'),p.createScript('return val+1'))(1), 2);
+  }, "Function constructor of all TrustedScripts works.");
+
   test(t => {
     assert_equals(new Function('return 1+1')(), 2);
   }, "Function constructor of string works.");
+
+  test(t => {
+    assert_equals(new Function('val', 'return val+1')(1), 2);
+  }, "Function constructor of all strings works.");
+
+  test(t => {
+    assert_equals(new Function('val', p.createScript('return val+1'))(1), 2);
+  }, "Function constructor of string and TrustedScript works.");
```

Comment:
Should we also add new tests for async function and the like here?

Reply:
Not super easy to do as you'd have to make the tests async, or deal with generators. I think that'd make the test flaky and unreliable (webkit cares about execution ordering).